core/src/main/java/org/hibernate/ogm/dialect/impl/AssociationContextImpl.java

```diff
@@ -25,21 +26,23 @@
 	private final AssociationTypeContext associationTypeContext;
 	private final OperationsQueue operationsQueue;
 	private final Tuple entityTuple;
+	private final TransactionCoordinator transactionCoordinator;
 
-	public AssociationContextImpl(AssociationTypeContext associationTypeContext, Tuple entityTuple) {
-		this( associationTypeContext, entityTuple, null );
+	public AssociationContextImpl(AssociationTypeContext associationTypeContext, Tuple entityTuple, TransactionCoordinator transactionCoordinator) {
+		this( associationTypeContext, entityTuple, null, transactionCoordinator );
 	}
 
 	public AssociationContextImpl(AssociationContextImpl original, OperationsQueue operationsQueue) {
-		this( original.associationTypeContext, original.entityTuple, operationsQueue );
+		this( original.associationTypeContext, original.entityTuple, operationsQueue, original.transactionCoordinator );
 	}
 
-	private AssociationContextImpl(AssociationTypeContext associationTypeContext, Tuple entityTuple, OperationsQueue operationsQueue) {
```

Comment:
it seems that the client for getTransactionCoordinator() will throw an NPE is we don't make sure thisnew _transactionCoordinator_ is never null. Add the assertion as the other parameter?